scripts/system_performance/system_performance_queries.py

```diff
@@ -0,0 +1,159 @@
+        # pylint: enable=line-too-long
+
+    @staticmethod
+    def query_sub_segment_replace(query_arguments):
+        """Select last query argument based on argument input - sub_segment selection"""
+        if not query_arguments['start_date']:
+            interval_range = "INTERVAL {} {}".format(query_arguments['interval'],
+                                                     query_arguments['time_scale'])
+
+            query_sub_segment = ">= DATE_SUB('{}', {})".format(query_arguments['end_date'],
+                                                               interval_range)
+        else:
+            # When both start and end date inputs are populated, query between those dates.
+            query_sub_segment = "BETWEEN '{}' AND '{}'".format(query_arguments['start_date'],
+                                                               query_arguments['end_date'])
+        return query_sub_segment
+
+    @staticmethod
+    def set_date_segment(start_date, end_date):
+        """Set date segment within query"""
+        if start_date == 'CURDATE()':
+            date_segment = 'CURDATE()'
+        else:
+            date_segment = end_date
+        return "= '{}'".format(date_segment)
+
+    def query_segment_replace(self, query_arguments):
+        """Handles the interchangeable segment of query to return either intervals of
+        time or period between two user specified dates and whether or not to
+        include a filter by retry run or not."""
+
+        returned_args = []
+
+        # If end date is None, query only for rows created on or up to current date
+        if query_arguments['start_date'] == query_arguments['end_date']:
+            query_segment = self.set_date_segment(start_date=query_arguments['start_date'],
+                                                  end_date=query_arguments['end_date'])
+            query_arguments['start_date'] = ''
+            returned_args.append(query_segment)
+        else:
+            # Determining which sub query segment to place in query.
+            query_segment = self.query_sub_segment_replace(query_arguments,)
+            returned_args.append(query_segment)
+
+        if query_arguments['instrument_id'] is not None:
+            # Applying instrument_id query argument segments when instrument_id argument
+            # populated as method arg
+            instrument_id_arg = ", instrument_id"
+            query_arguments['instrument_id'] = ', {}'.format(query_arguments['instrument_id'])
+        else:
+            query_arguments['instrument_id'] = ''
+            instrument_id_arg = ''
+
+        returned_args.append(instrument_id_arg)  # Appending instrument_id_arg
+        print(returned_args)
+        return [returned_args]
+
+    # pylint: disable=too-many-arguments
+    def get_data_by_status_over_time(self, selection='run_number', status_id=4, retry_run='',
+                                     anomic_aphasia='finished', end_date='CURDATE()', interval=1,
```

Comment:
need to change this variable name `anomic_aphasia`

Reply:
😢